pulsar/producer_partition.go

```diff
@@ -237,6 +239,11 @@ func (p *partitionProducer) grabCnx() error {
 		Schema:                   pbSchema,
 		Epoch:                    proto.Uint64(atomic.LoadUint64(&p.epoch)),
 		UserProvidedProducerName: proto.Bool(p.userProvidedProducerName),
+		ProducerAccessMode:       toProtoProducerAccessMode(p.options.ProducerAccessMode).Enum(),
+	}
+
+	if p.topicEpoch > -1 {
```

Comment:
The default value should be 0, not -1.

nit: You should not use `> n` on a unsigned integer, even if you converted it to a signed integer. For example, 0xFFFFFFFFFFFFFFFFFFFF should be a valid `uint64` epoch (though in Java it's overflowed), but when it's converted to `int64`, it will be -1 and treated as negative. I think you'd better use `uint64` and avoid type cast.

https://github.com/apache/pulsar-client-go/blob/d9b18d0690c15d5da94c9f324d9c2618e3d60bfd/pulsar/internal/pulsar_proto/PulsarApi.pb.go#L3518-L3522

Reply:
> The default value should be 0, not -1.

Because Go natively does not support `optional`. We need to initialize it to `-1` to judge whether have received `topicEpoch` from the broker(The broker is incremented from 0). 

> nit: You should not use > n on a unsigned integer, even if you converted it to a signed integer. For example, 0xFFFFFFFFFFFFFFFFFFFF should be a valid uint64 epoch (though in Java it's overflowed), but when it's converted to int64, it will be -1 and treated as negative. I think you'd better use uint64 and avoid type cast.

I use uint64 ptr instead of it.